File: sessions/week-00-introduction/algorithms/java/ReverseString.java
public class ReverseString {
    public String reverseString(String input) {
        char[] chars = input.toCharArray();
        int left = 0, right = chars.length - 1;

        while (left < right) {
            char temp = chars[left];
            chars[left] = chars[right];
            chars[right] = temp;
            left++;
            right--;
        }

        return new String(chars);
    }

    public static void main(String[] args) {
        ReverseString rs = new ReverseString();
        System.out.println(rs.reverseString("hello")); // Output: "olleh"
    }
}

File: sessions/week-00-introduction/algorithms/javascript/reverse_string.js
function reverseString(input) {
    const chars = input.split('');
    let left = 0, right = chars.length - 1;

    while (left < right) {
        const temp = chars[left];
        chars[left] = chars[right];
        chars[right] = temp;
        left++;
        right--;
    }

    return chars.join('');
}

// Example:
console.log(reverseString("hello")); // Output: "olleh"
